### PR TITLE
[PXZ-73] Implement POST, PUT and DELETE endpoints for products

### PR DESCRIPTION
**Pull request by juliabaptista**
# [PXZ-73] Implement POST, PUT and DELETE endpoints for products

# Product Management

## Technical Tasks:
- [x] Implement POST endpoint
- [x] Implement PUT endpoint
- [x] Implement DELETE endpoint
- [x] Validation and error handling for product

## Postman outputs of products:

> **POST endpoint - add product** <br />

Screenshot 2024-02-16 at 18 38 20 <br />
<br />

> **PUT endpoint - update product** <br />

Screenshot 2024-02-16 at 18 42 08 <br />
<br />

> **DELETE endpoint - archive product** <br />

Screenshot 2024-02-16 at 18 44 41 <br />
<br />

> **Error handling: product side** <br />

Screenshot 2024-02-16 at 18 38 55

Screenshot 2024-02-16 at 18 42 55

Screenshot 2024-02-16 at 18 45 04 <br />
<br />

> **Error handling: user permission side** <br />

Screenshot 2024-02-17 at 15 21 17

Screenshot 2024-02-17 at 15 21 38

Screenshot 2024-02-17 at 15 21 53

[PXZ-73]: https://jbptst.atlassian.net/browse/PXZ-73?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/de/zalando/controller/ProductController.java [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 @RestController
 @RequestMapping("/products")
 @RequiredArgsConstructor
 public class ProductController {
 
   ProductService productService;
```
Comment (f4r33) [inline src/main/java/de/zalando/controller/ProductController.java:38]: private final

File: src/main/java/de/zalando/controller/ProductController.java [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
-
+  UserService userService;
```
Comment (f4r33) [inline src/main/java/de/zalando/controller/ProductController.java:39]: private final

File: src/main/java/de/zalando/controller/ProductController.java [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
   @Autowired
-  public ProductController(ProductService productService) {
+  public ProductController(ProductService productService, UserService userService) {
```
Comment (f4r33) [inline src/main/java/de/zalando/controller/ProductController.java:41]: remove constructor

File: src/main/java/de/zalando/service/ProductService.java [hunk 4]
```diff
@@ -1,15 +1,21 @@
 package de.zalando.service;
 
+import de.zalando.dto.ProductRequest;
+import de.zalando.exception.DuplicateProductException;
+<<<<<<< HEAD
```
Comment (f4r33) [inline src/main/java/de/zalando/service/ProductService.java:5]: Fix conflict

**Review by f4r33 — CHANGES_REQUESTED** [review]
Fix conflict

File: src/main/java/de/zalando/controller/ProductController.java (hunk 2 above)
Comment (juliabaptista) [reply src/main/java/de/zalando/controller/ProductController.java:39]: @faruka-startsteps I changed MULTIPLE times and I don't know why it's still like that 😂

File: src/main/java/de/zalando/controller/ProductController.java (hunk 1 above)
Comment (juliabaptista) [reply src/main/java/de/zalando/controller/ProductController.java]: @faruka-startsteps I changed MULTIPLE times and I don't know why it's still like that 😂

File: src/main/java/de/zalando/controller/ProductController.java (hunk 3 above)
Comment (juliabaptista) [reply src/main/java/de/zalando/controller/ProductController.java]: @faruka-startsteps same 😂

**Review by faruka-startsteps — APPROVED** [review]

**Pull request merged**